static/panes/executor.js

```diff
@@ -898,22 +907,32 @@ Executor.prototype.initCallbacks = function () {
     }
 };
 
+Executor.prototype.shouldEmitExecutionOnFieldChange = function () {
+    return this.settings.executorCompileOnChange;
+};
+
 Executor.prototype.onOptionsChange = function (options) {
-    this.options = options;
-    this.saveState();
-    this.compile();
+    if (this.shouldEmitExecutionOnFieldChange()) {
+        this.options = options;
+        this.saveState();
```

Comment:
should we unconditionally note options and save state? And only `compile()` if the flag is set?